src/Npgsql/NpgsqlParameterCollection.cs

```diff
@@ -480,6 +480,7 @@ public override void Clear()
 
             InternalList.Clear();
             LookupClear();
+            PlaceholderType = PlaceholderType.NoParameters;
```

Comment:
Looks good @zitmen, thanks. Can you please add a regression test in CommandTests?

Reply:
@roji the test has been added